### PR TITLE
Fixes for onnx==1.11.0

### PR DESCRIPTION
**Pull request by xadupre**
* remove many lines of codes introduced in the early stages of the development and not needed anymore
* introduce TARGET_OPSET_ML for testing purpose
* use TARGET_OPSET to enable, disable unittest instead of onnx.__version__

### CODE REVIEW

**Comment by BowenBao** [comment]
Thanks @xadupre for doing this check, I can follow up on this if you want. 
The CI for onnx 1.11-rc seems to need installing `wget` beforehand.

**Comment by xadupre** [comment]
> Thanks @xadupre for doing this check, I can follow up on this if you want. The CI for onnx 1.11-rc seems to need installing `wget` beforehand.

It seems so. That's a surprising dependency.

File: tests/test_algebra_complex.py [hunk 1]
```diff
@@ -21,8 +19,7 @@ class TestAlgebraComplex(unittest.TestCase):
 
     @unittest.skipIf(Complex64TensorType is None,
                      reason="not available")
-    @unittest.skipIf(StrictVersion(onnx.__version__) < StrictVersion('1.8.0'),
-                     reason="not implemented")
+    @unittest.skipIf(TARGET_OPSET < 13, reason="not implemented")
```
Comment (BowenBao) [inline tests/test_algebra_complex.py:22]: Similar to above should it be  ` <14`？

File: tests/test_algebra_converters.py [hunk 2]
```diff
@@ -18,8 +16,7 @@
 
 class TestAlgebraConverters(unittest.TestCase):
 
-    @unittest.skipIf(StrictVersion(onnx.__version__) < StrictVersion("1.4.0"),
-                     reason="not available")
+    @unittest.skipIf(TARGET_OPSET < 10, reason="not available")
```
Comment (BowenBao) [inline tests/test_algebra_converters.py:19]: ditto

File: tests/test_sklearn_tfidf_vectorizer_converter.py [hunk 3]
```diff
@@ -22,22 +22,22 @@
...
 from skl2onnx import convert_sklearn
 from skl2onnx.common.data_types import StringTensorType, FloatTensorType
-import onnx
-import onnxruntime
+from onnxruntime import __version__ as ort_version
 from onnxruntime import InferenceSession
 from test_utils import dump_data_and_model, TARGET_OPSET
 
 
+ort_version = '.'.join(ort_version.split('.')[:2])
```
Comment (BowenBao) [inline tests/test_sklearn_tfidf_vectorizer_converter.py:30]: nit: this appears multiple time, maybe put into test_utils?

File: tests/test_algebra_cascade.py [hunk 4]
```diff
@@ -25,8 +23,7 @@
 
 class TestOnnxOperatorsCascade(unittest.TestCase):
 
-    @unittest.skipIf(StrictVersion(onnx.__version__) < StrictVersion("1.4.0"),
```
[the remark this answers is not in the document]
Comment (BowenBao) [reply tests/test_algebra_cascade.py:28]: update: after looking at more changes, it seems the new opset version should be more accurate. Feel free to resolve if that's the case.

**Review by BowenBao — APPROVED** [review]
Thanks for the updates. Looks good. Left a few questions checking if my understanding is correct.

File: tests/test_sklearn_adaboost_converter.py [hunk 5]
```diff
@@ -72,21 +61,13 @@ def test_ada_boost_classifier_samme_r_decision_function(self):
         )
         self.assertIsNotNone(model_onnx)
         dump_data_and_model(
-            X_test,
+            X_test[:5],
```
[the remark this answers is not in the document]
Comment (xadupre) [reply tests/test_sklearn_adaboost_converter.py:64]: I forgot to remove it. I did it after adding verbose=True to understand the failure.

File: tests/test_sklearn_tfidf_vectorizer_converter.py (hunk 3 above)
Comment (xadupre) [reply tests/test_sklearn_tfidf_vectorizer_converter.py:30]: Then the code can work with onnxruntime-training as well.

File: tests/test_algebra_complex.py (hunk 1 above)
Comment (xadupre) [reply tests/test_algebra_complex.py:22]: See https://github.com/onnx/onnx/blob/main/docs/Versioning.md.

**Pull request merged**